src/main/scala/edu/twitter/model/impl/neuralnetwork/NeuralNetworkBuilder.scala

```diff
@@ -0,0 +1,118 @@
+package edu.twitter.model.impl.neuralnetwork
+
+import java.io._
+
+import edu.twitter.model.api.{GenericModel, GenericModelBuilder}
+import edu.twitter.model.impl.TweetsLoader
+import org.apache.spark.SparkContext
+import org.deeplearning4j.eval.Evaluation
+import org.deeplearning4j.models.embeddings.loader.WordVectorSerializer
+import org.deeplearning4j.nn.conf.layers.{GravesLSTM, RnnOutputLayer}
+import org.deeplearning4j.nn.conf.{GradientNormalization, MultiLayerConfiguration, NeuralNetConfiguration, Updater}
+import org.deeplearning4j.nn.multilayer.MultiLayerNetwork
+import org.deeplearning4j.nn.weights.WeightInit
+import org.deeplearning4j.util.ModelSerializer
+import org.nd4j.linalg.activations.Activation
+import org.nd4j.linalg.lossfunctions.LossFunctions
+
+/**
+  * Build and evaluate the Neural network model from training and testing data set.
+  */
+class NeuralNetworkBuilder(sc: SparkContext) extends GenericModelBuilder {
+
+  val modelPath = this.getClass().getClassLoader().getResource("saved-models").getPath() + File.separator + "NeuralNetworkModel.net"
+
+  /** Location (local file system) for the Google News vectors. */
+  //val WORD_VECTORS_PATH: String = this.getClass.getClassLoader.getResource("NewsModel.txt").getPath
+  val WORD_VECTORS_PATH: String = this.getClass.getClassLoader.getResource("GoogleNews-vectors-negative300.bin.gz").getPath
+
+  //val wordVectors = WordVectorSerializer.loadTxtVectors(new File(WORD_VECTORS_PATH))
+  val wordVectors = WordVectorSerializer.readWord2VecModel(new File(WORD_VECTORS_PATH))
+  val vectorSize: Int = wordVectors.getWordVector(wordVectors.vocab.wordAtIndex(0)).length // 100 in our case
+
+  /**
+    * Run the recipe responsible for constructing the model.
+    *
+    * @return an instance of generic model.
+    */
+  override def build(): GenericModel = {
+
+    if (checkModelExist()) {
+      val model = ModelSerializer.restoreMultiLayerNetwork(modelPath)
+      return new NeuralNetworkModel(model, wordVectors)
+    }
+
+    val batchSize = 256 //Number of examples in each minibatch
+    val nEpochs = 1 //Number of epochs (full passes of training data) to train on
+    val truncateReviewsToLength = 280 //Truncate reviews with length (# words) greater than this
+
+    //DataSetIterators for training and testing respectively
+    val dataPath = this.getClass.getClassLoader.getResource("labeled-tweets").getPath
+    val data = new TweetsLoader(sc).loadDataSet(dataPath)
+    val Array(trainData, testData) = data.randomSplit(Array(0.7, 0.3))
+    val train = new DataIterator(trainData, wordVectors, batchSize, truncateReviewsToLength)
+    val test = new DataIterator(testData, wordVectors, batchSize, truncateReviewsToLength)
+
+    //Set up network configuration
+    val conf: MultiLayerConfiguration = new NeuralNetConfiguration.Builder()
```

Comment:
extract building the configuration to a method please, something like
``` scala
val conf = buildConfig
```

Reply:
Done 👍